At the very beginning, I import everything we need. In this care, I import pandas as the standard "pd" in order to clean and wrangle the data.

In [396]:
import pandas as pd


Using the pd function .read_csv(), my csv files are made readable by pandas. 

In [397]:
student_life_df = pd.read_csv('../Data/AI_Impact_Student_Life_2026.csv')
jobs_df = pd.read_csv('../Data/ai_job_impact.csv')
content_df = pd.read_csv('../Data/Global_AI_Content_Impact_Dataset.csv')

For an initial overlook of the data and how it is shaped, I use .head() and .tail() to see the first and last 5 rows, unless otherwise specified. .info() tells me the datatypes of each column as well as how many values are null (if any). Lastly, .shape gives me the dimensions of the table, rows and columns. 

In [446]:
def intro_report(list_of_df):
    for df in list_of_df:
        print("Overview of Dataframes")
        print("First 5 Rows:")
        print(df.head())
        print("Last 5 Rows:")
        print(df.tail())
        print('Nulls and Data Types:')
        print(df.info())
        print('Rows by Columns')
        print (df.shape)

intro_report([student_life_df, content_df, jobs_df])

Overview of Dataframes
First 5 Rows:
  Student_ID  Age                 Major Primary_AI_Tool  Task_Frequency_Daily  \
0   STU-6019   23  Software Engineering      Gemini Pro                     1   
1   STU-6962   22        Modern History  GitHub Copilot                     3   
2   STU-2338   18          Data Science      Perplexity                     2   
3   STU-1380   19               Biology          Claude                     5   
4   STU-1837   19               Biology         ChatGPT                    10   

     Main_Usage_Case  GPA_Baseline  GPA_Post_AI  Time_Saved_Hours_Weekly  \
0     Code Debugging          2.62         2.62                        9   
1     Essay Drafting          3.99         4.00                        7   
2  Literature Review          2.57         2.57                       15   
3     Essay Drafting          2.67         2.87                       12   
4     Code Debugging          3.65         3.85                        5   

  AI_Ethics_Concern

In [398]:
student_life_df.head()

,Student_ID,Age,Major,Primary_AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
0,STU-6019,23,Software Engineering,Gemini Pro,1,Code Debugging,2.62,2.62,9,Medium,3
1,STU-6962,22,Modern History,GitHub Copilot,3,Essay Drafting,3.99,4.00,7,Medium,4
2,STU-2338,18,Data Science,Perplexity,2,Literature Review,2.57,2.57,15,High,7
3,STU-1380,19,Biology,Claude 3.5,5,Essay Drafting,2.67,2.87,12,Low,5
4,STU-1837,19,Biology,ChatGPT-4o,10,Code Debugging,3.65,3.85,5,High,9


In [399]:
student_life_df.tail(10)

,Student_ID,Age,Major,Primary_AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
1490,STU-0866,23,Fine Arts,GitHub Copilot,7,Code Debugging,3.60,3.80,15,Medium,3
1491,STU-4535,18,Software Engineering,Gemini Pro,2,Brainstorming,3.73,3.93,9,Medium,5
1492,STU-4908,24,Software Engineering,GitHub Copilot,9,Essay Drafting,3.14,3.44,6,Medium,9
1493,STU-2710,25,Modern History,Gemini Pro,2,Essay Drafting,2.74,2.74,6,Low,1
1494,STU-5328,20,Fine Arts,Gemini Pro,5,Code Debugging,3.36,3.56,14,High,3
1495,STU-2711,21,Modern History,GitHub Copilot,4,Brainstorming,3.35,3.55,7,Medium,7
1496,STU-4888,21,Biology,GitHub Copilot,3,Essay Drafting,3.59,3.89,13,Medium,7
1497,STU-0180,23,Software Engineering,GitHub Copilot,8,Code Debugging,3.20,3.40,7,Low,2
1498,STU-4392,23,Fine Arts,Perplexity,6,Brainstorming,3.99,4.00,15,Low,7
1499,STU-4702,23,Data Science,Gemini Pro,4,Exam Prep,3.53,3.43,3,Medium,8


In [400]:
student_life_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 1500 entries, 0 to 1499
Data columns (total 11 columns):
 #   Column                   Non-Null Count  Dtype  
---  ------                   --------------  -----  
 0   Student_ID               1500 non-null   str    
 1   Age                      1500 non-null   int64  
 2   Major                    1500 non-null   str    
 3   Primary_AI_Tool          1500 non-null   str    
 4   Task_Frequency_Daily     1500 non-null   int64  
 5   Main_Usage_Case          1500 non-null   str    
 6   GPA_Baseline             1500 non-null   float64
 7   GPA_Post_AI              1500 non-null   float64
 8   Time_Saved_Hours_Weekly  1500 non-null   int64  
 9   AI_Ethics_Concern        1500 non-null   str    
 10  Career_Confidence_Score  1500 non-null   int64  
dtypes: float64(2), int64(4), str(5)
memory usage: 129.0 KB


In [401]:
student_life_df.shape

(1500, 11)

In [402]:
jobs_df.head()

,Employee_ID,Age,Gender,Education_Level,Industry,Job_Role,Years_Experience,AI_Adoption_Level,Automation_Risk,Upskilling_Required,Salary_Before_AI,Salary_After_AI,Job_Status,Work_Hours_Per_Week,Remote_Work,Job_Satisfaction,Productivity_Change_%
0,E0001,50,Female,Bachelor,Marketing,Content Creator,26,High,High,Yes,106820,95455,Replaced,45,No,5,-10.64
1,E0002,45,Male,High School,Manufacturing,Quality Inspector,19,Low,Low,Yes,74131,72013,Unchanged,36,Yes,6,19.05
2,E0003,51,Female,Master,IT,DevOps Engineer,28,Medium,Medium,Yes,35311,42290,Modified,46,Yes,3,17.05
3,E0004,48,Male,PhD,Education,Teacher,24,Medium,Medium,Yes,114478,107820,Modified,50,No,9,-2.47
4,E0005,24,Male,Bachelor,Healthcare,Doctor,0,High,Medium,No,33890,40945,Modified,52,Yes,6,7.03


In [403]:
jobs_df.tail()

,Employee_ID,Age,Gender,Education_Level,Industry,Job_Role,Years_Experience,AI_Adoption_Level,Automation_Risk,Upskilling_Required,Salary_Before_AI,Salary_After_AI,Job_Status,Work_Hours_Per_Week,Remote_Work,Job_Satisfaction,Productivity_Change_%
1995,E1996,55,Other,PhD,Healthcare,Technician,30,Medium,Low,Yes,99082,101090,Unchanged,52,Yes,7,25.67
1996,E1997,31,Male,PhD,Finance,Investment Analyst,9,High,Medium,Yes,32379,44653,Modified,52,Yes,8,9.38
1997,E1998,43,Female,Bachelor,Retail,Cashier,17,High,High,No,53424,74715,Modified,49,Yes,6,-0.57
1998,E1999,43,Female,Bachelor,Marketing,Digital Marketer,20,Medium,Low,No,57821,68499,Unchanged,41,No,9,9.31
1999,E2000,23,Other,Master,Marketing,SEO Specialist,0,High,High,Yes,33830,34809,Modified,37,Yes,9,24.40


In [404]:
jobs_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 2000 entries, 0 to 1999
Data columns (total 17 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Employee_ID            2000 non-null   str    
 1   Age                    2000 non-null   int64  
 2   Gender                 2000 non-null   str    
 3   Education_Level        2000 non-null   str    
 4   Industry               2000 non-null   str    
 5   Job_Role               2000 non-null   str    
 6   Years_Experience       2000 non-null   int64  
 7   AI_Adoption_Level      2000 non-null   str    
 8   Automation_Risk        2000 non-null   str    
 9   Upskilling_Required    2000 non-null   str    
 10  Salary_Before_AI       2000 non-null   int64  
 11  Salary_After_AI        2000 non-null   int64  
 12  Job_Status             2000 non-null   str    
 13  Work_Hours_Per_Week    2000 non-null   int64  
 14  Remote_Work            2000 non-null   str    
 15  Job_Satisfactio

In [405]:
jobs_df.shape

(2000, 17)

In [406]:
content_df.head()

,Country,Year,Industry,AI Adoption Rate (%),AI-Generated Content Volume (TBs per year),Job Loss Due to AI (%),Revenue Increase Due to AI (%),Human-AI Collaboration Rate (%),Top AI Tools Used,Regulation Status,Consumer Trust in AI (%),Market Share of AI Companies (%)
0,South Korea,2022,Media,44.29,33.09,16.77,46.12,74.79,Bard,Strict,40.77,18.73
1,China,2025,Legal,34.75,66.74,46.89,52.46,26.17,DALL-E,Strict,35.67,35.02
2,USA,2022,Automotive,81.06,96.13,10.66,45.60,39.66,Stable Diffusion,Moderate,54.47,22.76
3,France,2021,Legal,85.24,93.76,27.70,78.24,29.45,Claude,Moderate,51.84,1.93
4,France,2021,Gaming,78.95,45.62,17.45,1.05,21.70,Midjourney,Strict,41.77,21.41


In [407]:
content_df.tail()

,Country,Year,Industry,AI Adoption Rate (%),AI-Generated Content Volume (TBs per year),Job Loss Due to AI (%),Revenue Increase Due to AI (%),Human-AI Collaboration Rate (%),Top AI Tools Used,Regulation Status,Consumer Trust in AI (%),Market Share of AI Companies (%)
195,Germany,2021,Automotive,89.44,52.98,48.47,12.14,30.60,DALL-E,Lenient,77.21,44.35
196,Germany,2020,Media,70.11,28.26,27.62,57.86,58.71,DALL-E,Strict,78.74,31.73
197,France,2023,Marketing,65.77,49.83,39.94,79.44,43.73,Synthesia,Lenient,81.58,14.62
198,Australia,2023,Automotive,45.35,20.49,33.21,50.50,41.73,Claude,Strict,47.42,43.11
199,South Korea,2020,Healthcare,10.53,20.97,23.64,34.27,45.67,Stable Diffusion,Moderate,58.52,33.37


In [408]:
content_df.info()

<class 'pandas.DataFrame'>
RangeIndex: 200 entries, 0 to 199
Data columns (total 12 columns):
 #   Column                                      Non-Null Count  Dtype  
---  ------                                      --------------  -----  
 0   Country                                     200 non-null    str    
 1   Year                                        200 non-null    int64  
 2   Industry                                    200 non-null    str    
 3   AI Adoption Rate (%)                        200 non-null    float64
 4   AI-Generated Content Volume (TBs per year)  200 non-null    float64
 5   Job Loss Due to AI (%)                      200 non-null    float64
 6   Revenue Increase Due to AI (%)              200 non-null    float64
 7   Human-AI Collaboration Rate (%)             200 non-null    float64
 8   Top AI Tools Used                           200 non-null    str    
 9   Regulation Status                           200 non-null    str    
 10  Consumer Trust in AI (%) 

In [409]:
content_df.shape

(200, 12)

I selected these two columns specifically so I could better see a broad overview of the effect of AI on GPA.

In [410]:
student_life_df[['GPA_Baseline', 'GPA_Post_AI']]

,GPA_Baseline,GPA_Post_AI
0,2.62,2.62
1,3.99,4.00
2,2.57,2.57
3,2.67,2.87
4,3.65,3.85
...,...,...
1495,3.35,3.55
1496,3.59,3.89
1497,3.20,3.40
1498,3.99,4.00


I decided to use boolean indexing here to filter with conditions to find the amount of students whos GPA grew, declined, and stayed the same. Here I can see that if we just look at GPA's, it will show that AI might be helping students get higher GPA's. 

In [411]:
GPA_growth_df = student_life_df[student_life_df['GPA_Baseline'] < student_life_df['GPA_Post_AI']]
GPA_growth_df.shape

(894, 11)

In [412]:
GPA_decline_df = student_life_df[student_life_df['GPA_Baseline'] > student_life_df['GPA_Post_AI']]
GPA_decline_df.shape

(301, 11)

In [413]:
GPA_same_df = student_life_df[student_life_df['GPA_Baseline'] == student_life_df['GPA_Post_AI']]
GPA_same_df.shape

(305, 11)

.describe() gives me the statistics of the values represented by integers. value_counts() tells me how many rows include a given value as a result. From these, I can glean that the most common age in this dataset is 21, the GPA does increase on average, and the most used AI tool is GitHub Copilot.

In [414]:
student_life_df.describe()

,Age,Task_Frequency_Daily,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,Career_Confidence_Score
count,1500.000000,1500.000000,1500.000000,1500.000000,1500.00000,1500.000000
mean,21.494000,5.407333,3.256853,3.344587,8.51000,5.417333
std,2.297277,2.905462,0.430583,0.437409,4.07148,2.844977
min,18.000000,1.000000,2.500000,2.400000,2.00000,1.000000
25%,20.000000,3.000000,2.880000,2.980000,5.00000,3.000000
50%,21.000000,5.000000,3.260000,3.360000,9.00000,5.000000
75%,23.000000,8.000000,3.620000,3.710000,12.00000,8.000000
max,25.000000,10.000000,4.000000,4.000000,15.00000,10.000000


In [415]:
student_life_df['Primary_AI_Tool'].value_counts()

Primary_AI_Tool
GitHub Copilot    311
Gemini Pro        302
Perplexity        302
Claude 3.5        297
ChatGPT-4o        288
Name: count, dtype: int64

In [416]:
content_df['Top AI Tools Used'].value_counts()

Top AI Tools Used
Midjourney          37
Claude              31
ChatGPT             31
Stable Diffusion    27
DALL-E              26
Bard                24
Synthesia           24
Name: count, dtype: int64

In [417]:
jobs_df.describe()

,Age,Years_Experience,Salary_Before_AI,Salary_After_AI,Work_Hours_Per_Week,Job_Satisfaction,Productivity_Change_%
count,2000.000000,2000.000000,2000.000000,2000.000000,2000.00000,2000.000000,2000.000000
mean,40.558000,16.663500,73942.072500,78428.642500,44.85100,6.020500,9.785820
std,10.786418,10.746675,26055.823793,29351.599013,5.71254,2.006263,17.187882
min,22.000000,0.000000,30036.000000,24447.000000,35.00000,3.000000,-19.990000
25%,32.000000,8.000000,51665.500000,54086.750000,40.00000,4.000000,-5.355000
50%,40.000000,16.000000,74620.000000,76820.500000,45.00000,6.000000,9.850000
75%,50.000000,26.000000,95418.250000,100730.250000,50.00000,8.000000,24.582500
max,59.000000,37.000000,119976.000000,161745.000000,54.00000,9.000000,39.990000


I was curious about how each employee would qualify in terms of years of experience, and this value would be easy to understand if used in a visual, so I created a function for showing specifically what a person's experience would lead them to apply for in the future. 

In [418]:
def categorize_experience(Years_Experience):
    if Years_Experience <= 2:
        return 'Entry Level'
    elif Years_Experience >= 3 and Years_Experience < 5:
        return 'Mid Level'
    else:
        return 'Senior'

jobs_df['ExperienceLevel'] = jobs_df['Years_Experience'].apply(categorize_experience)

In [419]:
jobs_df.head()

,Employee_ID,Age,Gender,Education_Level,Industry,Job_Role,Years_Experience,AI_Adoption_Level,Automation_Risk,Upskilling_Required,Salary_Before_AI,Salary_After_AI,Job_Status,Work_Hours_Per_Week,Remote_Work,Job_Satisfaction,Productivity_Change_%,ExperienceLevel
0,E0001,50,Female,Bachelor,Marketing,Content Creator,26,High,High,Yes,106820,95455,Replaced,45,No,5,-10.64,Senior
1,E0002,45,Male,High School,Manufacturing,Quality Inspector,19,Low,Low,Yes,74131,72013,Unchanged,36,Yes,6,19.05,Senior
2,E0003,51,Female,Master,IT,DevOps Engineer,28,Medium,Medium,Yes,35311,42290,Modified,46,Yes,3,17.05,Senior
3,E0004,48,Male,PhD,Education,Teacher,24,Medium,Medium,Yes,114478,107820,Modified,50,No,9,-2.47,Senior
4,E0005,24,Male,Bachelor,Healthcare,Doctor,0,High,Medium,No,33890,40945,Modified,52,Yes,6,7.03,Entry Level


In [ ]:
def categorize_adoption_rate(adoption_rate):
    if adoption_rate <= 33.33:
        return 'Low'
    elif adoption_rate >= 33.34 and adoption_rate <= 66.66:
        return 'Medium'
    else:
        return 'High'

content_df['AI_Adoption_Level'] = content_df['AI Adoption Rate (%)'].apply(categorize_adoption_rate)


In [421]:
content_df

,Country,Year,Industry,AI Adoption Rate (%),AI-Generated Content Volume (TBs per year),Job Loss Due to AI (%),Revenue Increase Due to AI (%),Human-AI Collaboration Rate (%),Top AI Tools Used,Regulation Status,Consumer Trust in AI (%),Market Share of AI Companies (%),AI_Adoption_Level
0,South Korea,2022,Media,44.29,33.09,16.77,46.12,74.79,Bard,Strict,40.77,18.73,Medium
1,China,2025,Legal,34.75,66.74,46.89,52.46,26.17,DALL-E,Strict,35.67,35.02,Medium
2,USA,2022,Automotive,81.06,96.13,10.66,45.60,39.66,Stable Diffusion,Moderate,54.47,22.76,High
3,France,2021,Legal,85.24,93.76,27.70,78.24,29.45,Claude,Moderate,51.84,1.93,High
4,France,2021,Gaming,78.95,45.62,17.45,1.05,21.70,Midjourney,Strict,41.77,21.41,High
...,...,...,...,...,...,...,...,...,...,...,...,...,...
195,Germany,2021,Automotive,89.44,52.98,48.47,12.14,30.60,DALL-E,Lenient,77.21,44.35,High
196,Germany,2020,Media,70.11,28.26,27.62,57.86,58.71,DALL-E,Strict,78.74,31.73,High
197,France,2023,Marketing,65.77,49.83,39.94,79.44,43.73,Synthesia,Lenient,81.58,14.62,Medium
198,Australia,2023,Automotive,45.35,20.49,33.21,50.50,41.73,Claude,Strict,47.42,43.11,Medium


This dataset did not require the use of .drop(), but I wanted to practice using it in this analysis.

In [422]:
jobs_df = jobs_df.drop('Gender', axis=1)
jobs_df

,Employee_ID,Age,Education_Level,Industry,Job_Role,Years_Experience,AI_Adoption_Level,Automation_Risk,Upskilling_Required,Salary_Before_AI,Salary_After_AI,Job_Status,Work_Hours_Per_Week,Remote_Work,Job_Satisfaction,Productivity_Change_%,ExperienceLevel
0,E0001,50,Bachelor,Marketing,Content Creator,26,High,High,Yes,106820,95455,Replaced,45,No,5,-10.64,Senior
1,E0002,45,High School,Manufacturing,Quality Inspector,19,Low,Low,Yes,74131,72013,Unchanged,36,Yes,6,19.05,Senior
2,E0003,51,Master,IT,DevOps Engineer,28,Medium,Medium,Yes,35311,42290,Modified,46,Yes,3,17.05,Senior
3,E0004,48,PhD,Education,Teacher,24,Medium,Medium,Yes,114478,107820,Modified,50,No,9,-2.47,Senior
4,E0005,24,Bachelor,Healthcare,Doctor,0,High,Medium,No,33890,40945,Modified,52,Yes,6,7.03,Entry Level
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1995,E1996,55,PhD,Healthcare,Technician,30,Medium,Low,Yes,99082,101090,Unchanged,52,Yes,7,25.67,Senior
1996,E1997,31,PhD,Finance,Investment Analyst,9,High,Medium,Yes,32379,44653,Modified,52,Yes,8,9.38,Senior
1997,E1998,43,Bachelor,Retail,Cashier,17,High,High,No,53424,74715,Modified,49,Yes,6,-0.57,Senior
1998,E1999,43,Bachelor,Marketing,Digital Marketer,20,Medium,Low,No,57821,68499,Unchanged,41,No,9,9.31,Senior


In order to maintain consistency between student_life_df['Primary_AI_Tool'] and content_df['Top AI Tools Used']. While we may lose a small amount of specificity, we won't be altering the statistics by doing so, since there isn't ChatGPT and ChatGPT-4o.

In [423]:
student_life_df['Primary_AI_Tool'] = student_life_df['Primary_AI_Tool'].replace({'Claude 3.5': 'Claude', 'ChatGPT-4o': 'ChatGPT'})
student_life_df['Primary_AI_Tool'].value_counts()

Primary_AI_Tool
GitHub Copilot    311
Gemini Pro        302
Perplexity        302
Claude            297
ChatGPT           288
Name: count, dtype: int64

isnull() and .sum() tells me which values if any are null/empty, and how many nulls there are.

In [424]:
student_life_df.isnull()

,Student_ID,Age,Major,Primary_AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
0,False,False,False,False,False,False,False,False,False,False,False
1,False,False,False,False,False,False,False,False,False,False,False
2,False,False,False,False,False,False,False,False,False,False,False
3,False,False,False,False,False,False,False,False,False,False,False
4,False,False,False,False,False,False,False,False,False,False,False
...,...,...,...,...,...,...,...,...,...,...,...
1495,False,False,False,False,False,False,False,False,False,False,False
1496,False,False,False,False,False,False,False,False,False,False,False
1497,False,False,False,False,False,False,False,False,False,False,False
1498,False,False,False,False,False,False,False,False,False,False,False


In [425]:
student_life_df.isnull().sum()

Student_ID                 0
Age                        0
Major                      0
Primary_AI_Tool            0
Task_Frequency_Daily       0
Main_Usage_Case            0
GPA_Baseline               0
GPA_Post_AI                0
Time_Saved_Hours_Weekly    0
AI_Ethics_Concern          0
Career_Confidence_Score    0
dtype: int64

.dtypes tells me what data types the columns hold, and if I wanted to change any of them I could use .astype(). For example: 

jobs_df['Productivity_Change_%'] = jobs_df['Productivity_Change_%'].astype(int64)

pd.to_datetime can be used to change dates/times to a specific datatype. If this dataset had a column to use this on, it would look like this:

jobs_df['Date'] = pd.to_datetime(jobs_df['Date'] format='%m/%d/%Y')

In [426]:
jobs_df.dtypes

Employee_ID                  str
Age                        int64
Education_Level              str
Industry                     str
Job_Role                     str
Years_Experience           int64
AI_Adoption_Level            str
Automation_Risk              str
Upskilling_Required          str
Salary_Before_AI           int64
Salary_After_AI            int64
Job_Status                   str
Work_Hours_Per_Week        int64
Remote_Work                  str
Job_Satisfaction           int64
Productivity_Change_%    float64
ExperienceLevel              str
dtype: object

I can tell with .duplicated() and .sum() that there are no dupliacted rows in this dataset. By using the function on just one column, one that I know will have duplicates, I can see that the 6 values in the Major column are repeated 1494 times. If there were a reason to drop duplicated, I would use .drop_duplicates()

In [427]:
duplicate_sum = student_life_df.duplicated().sum()

print(duplicate_sum)

0


In [428]:
duplicate_sum = student_life_df.duplicated(subset='Major').sum()

print(duplicate_sum)

1494


I had purposefully named the Experience Level column something just outside of the naming convention already used so that I could demonstrate the .rename function.

In [429]:
content_df.rename(columns={'Top AI Tools Used': 'AI_Tool'}, inplace=True)
content_df

,Country,Year,Industry,AI Adoption Rate (%),AI-Generated Content Volume (TBs per year),Job Loss Due to AI (%),Revenue Increase Due to AI (%),Human-AI Collaboration Rate (%),AI_Tool,Regulation Status,Consumer Trust in AI (%),Market Share of AI Companies (%),AI_Adoption_Level
0,South Korea,2022,Media,44.29,33.09,16.77,46.12,74.79,Bard,Strict,40.77,18.73,Medium
1,China,2025,Legal,34.75,66.74,46.89,52.46,26.17,DALL-E,Strict,35.67,35.02,Medium
2,USA,2022,Automotive,81.06,96.13,10.66,45.60,39.66,Stable Diffusion,Moderate,54.47,22.76,High
3,France,2021,Legal,85.24,93.76,27.70,78.24,29.45,Claude,Moderate,51.84,1.93,High
4,France,2021,Gaming,78.95,45.62,17.45,1.05,21.70,Midjourney,Strict,41.77,21.41,High
...,...,...,...,...,...,...,...,...,...,...,...,...,...
195,Germany,2021,Automotive,89.44,52.98,48.47,12.14,30.60,DALL-E,Lenient,77.21,44.35,High
196,Germany,2020,Media,70.11,28.26,27.62,57.86,58.71,DALL-E,Strict,78.74,31.73,High
197,France,2023,Marketing,65.77,49.83,39.94,79.44,43.73,Synthesia,Lenient,81.58,14.62,Medium
198,Australia,2023,Automotive,45.35,20.49,33.21,50.50,41.73,Claude,Strict,47.42,43.11,Medium


In [430]:
student_life_df.rename(columns={'Primary_AI_Tool': 'AI_Tool'})


,Student_ID,Age,Major,AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
0,STU-6019,23,Software Engineering,Gemini Pro,1,Code Debugging,2.62,2.62,9,Medium,3
1,STU-6962,22,Modern History,GitHub Copilot,3,Essay Drafting,3.99,4.00,7,Medium,4
2,STU-2338,18,Data Science,Perplexity,2,Literature Review,2.57,2.57,15,High,7
3,STU-1380,19,Biology,Claude,5,Essay Drafting,2.67,2.87,12,Low,5
4,STU-1837,19,Biology,ChatGPT,10,Code Debugging,3.65,3.85,5,High,9
...,...,...,...,...,...,...,...,...,...,...,...
1495,STU-2711,21,Modern History,GitHub Copilot,4,Brainstorming,3.35,3.55,7,Medium,7
1496,STU-4888,21,Biology,GitHub Copilot,3,Essay Drafting,3.59,3.89,13,Medium,7
1497,STU-0180,23,Software Engineering,GitHub Copilot,8,Code Debugging,3.20,3.40,7,Low,2
1498,STU-4392,23,Fine Arts,Perplexity,6,Brainstorming,3.99,4.00,15,Low,7


In [431]:
jobs_df.rename(columns={'ExperienceLevel': 'Experience_Level'}, inplace=True)
jobs_df

,Employee_ID,Age,Education_Level,Industry,Job_Role,Years_Experience,AI_Adoption_Level,Automation_Risk,Upskilling_Required,Salary_Before_AI,Salary_After_AI,Job_Status,Work_Hours_Per_Week,Remote_Work,Job_Satisfaction,Productivity_Change_%,Experience_Level
0,E0001,50,Bachelor,Marketing,Content Creator,26,High,High,Yes,106820,95455,Replaced,45,No,5,-10.64,Senior
1,E0002,45,High School,Manufacturing,Quality Inspector,19,Low,Low,Yes,74131,72013,Unchanged,36,Yes,6,19.05,Senior
2,E0003,51,Master,IT,DevOps Engineer,28,Medium,Medium,Yes,35311,42290,Modified,46,Yes,3,17.05,Senior
3,E0004,48,PhD,Education,Teacher,24,Medium,Medium,Yes,114478,107820,Modified,50,No,9,-2.47,Senior
4,E0005,24,Bachelor,Healthcare,Doctor,0,High,Medium,No,33890,40945,Modified,52,Yes,6,7.03,Entry Level
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
1995,E1996,55,PhD,Healthcare,Technician,30,Medium,Low,Yes,99082,101090,Unchanged,52,Yes,7,25.67,Senior
1996,E1997,31,PhD,Finance,Investment Analyst,9,High,Medium,Yes,32379,44653,Modified,52,Yes,8,9.38,Senior
1997,E1998,43,Bachelor,Retail,Cashier,17,High,High,No,53424,74715,Modified,49,Yes,6,-0.57,Senior
1998,E1999,43,Bachelor,Marketing,Digital Marketer,20,Medium,Low,No,57821,68499,Unchanged,41,No,9,9.31,Senior


In [432]:
student_life_df.sort_values(by=['Age', 'Major'], ascending=[True, False])

,Student_ID,Age,Major,Primary_AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
75,STU-3046,18,Software Engineering,Gemini Pro,2,Exam Prep,3.18,3.18,8,Medium,7
134,STU-5360,18,Software Engineering,Gemini Pro,8,Exam Prep,3.48,3.48,3,Medium,5
157,STU-5053,18,Software Engineering,GitHub Copilot,5,Exam Prep,2.60,2.70,9,High,6
159,STU-3925,18,Software Engineering,Gemini Pro,7,Literature Review,3.23,3.33,3,Medium,2
172,STU-2366,18,Software Engineering,GitHub Copilot,5,Code Debugging,2.54,2.44,9,Medium,4
...,...,...,...,...,...,...,...,...,...,...,...
1330,STU-1586,25,Biology,GitHub Copilot,9,Code Debugging,2.67,2.87,3,High,3
1355,STU-3395,25,Biology,Perplexity,10,Brainstorming,3.45,3.75,9,High,6
1373,STU-6943,25,Biology,Gemini Pro,2,Code Debugging,3.21,3.41,3,High,3
1395,STU-4929,25,Biology,Claude,4,Essay Drafting,3.84,3.84,3,Low,10


In [433]:
student_life_df.sort_index(ascending = False)

,Student_ID,Age,Major,Primary_AI_Tool,Task_Frequency_Daily,Main_Usage_Case,GPA_Baseline,GPA_Post_AI,Time_Saved_Hours_Weekly,AI_Ethics_Concern,Career_Confidence_Score
1499,STU-4702,23,Data Science,Gemini Pro,4,Exam Prep,3.53,3.43,3,Medium,8
1498,STU-4392,23,Fine Arts,Perplexity,6,Brainstorming,3.99,4.00,15,Low,7
1497,STU-0180,23,Software Engineering,GitHub Copilot,8,Code Debugging,3.20,3.40,7,Low,2
1496,STU-4888,21,Biology,GitHub Copilot,3,Essay Drafting,3.59,3.89,13,Medium,7
1495,STU-2711,21,Modern History,GitHub Copilot,4,Brainstorming,3.35,3.55,7,Medium,7
...,...,...,...,...,...,...,...,...,...,...,...
4,STU-1837,19,Biology,ChatGPT,10,Code Debugging,3.65,3.85,5,High,9
3,STU-1380,19,Biology,Claude,5,Essay Drafting,2.67,2.87,12,Low,5
2,STU-2338,18,Data Science,Perplexity,2,Literature Review,2.57,2.57,15,High,7
1,STU-6962,22,Modern History,GitHub Copilot,3,Essay Drafting,3.99,4.00,7,Medium,4


I used the dataframes I made before to see which Primary AI Tool had the worst/best post-AI GPA on average.

In [434]:
GPA_growth_df.groupby('Primary_AI_Tool')['GPA_Post_AI'].mean()

Primary_AI_Tool
ChatGPT-4o        3.439773
Claude 3.5        3.458671
Gemini Pro        3.486872
GitHub Copilot    3.426902
Perplexity        3.433352
Name: GPA_Post_AI, dtype: float64

In [435]:
GPA_decline_df.groupby('Primary_AI_Tool')['GPA_Post_AI'].mean()

Primary_AI_Tool
ChatGPT-4o        3.217647
Claude 3.5        3.042069
Gemini Pro        3.193913
GitHub Copilot    3.109219
Perplexity        3.161017
Name: GPA_Post_AI, dtype: float64

In [436]:
round(GPA_growth_df.groupby(['Major', 'Main_Usage_Case'])['GPA_Post_AI'].mean(), 2)

Major                    Main_Usage_Case  
Biology                  Brainstorming        3.58
                         Code Debugging       3.51
                         Essay Drafting       3.54
                         Exam Prep            3.50
                         Literature Review    3.36
Business Administration  Brainstorming        3.37
                         Code Debugging       3.47
                         Essay Drafting       3.44
                         Exam Prep            3.45
                         Literature Review    3.43
Data Science             Brainstorming        3.52
                         Code Debugging       3.42
                         Essay Drafting       3.36
                         Exam Prep            3.51
                         Literature Review    3.38
Fine Arts                Brainstorming        3.36
                         Code Debugging       3.45
                         Essay Drafting       3.39
                         Exam Prep     

In [437]:
jobs_df.groupby('Industry')['Productivity_Change_%'].agg(['min', 'mean', 'max'])

,min,mean,max
Industry,,,
Education,-19.99,10.014505,39.83
Finance,-19.58,9.565767,39.90
Healthcare,-19.88,9.307259,39.82
IT,-19.91,9.061074,39.99
Manufacturing,-19.93,10.697390,39.97
Marketing,-19.77,10.332527,39.91
Retail,-19.92,9.512878,39.86


In [438]:
student_life_df.groupby(['Primary_AI_Tool', 'Main_Usage_Case']).agg({
    'Time_Saved_Hours_Weekly': 'mean',
    'Task_Frequency_Daily': 'count'

})

Time_Saved_Hours_Weekly  \
Primary_AI_Tool Main_Usage_Case                              
ChatGPT         Brainstorming                     8.038462   
                Code Debugging                    7.897959   
                Essay Drafting                    8.666667   
                Exam Prep                         8.376812   
                Literature Review                 8.545455   
Claude          Brainstorming                     8.826923   
                Code Debugging                    8.604167   
                Essay Drafting                    8.490909   
                Exam Prep                         7.970588   
                Literature Review                 7.810811   
Gemini Pro      Brainstorming                     8.553846   
                Code Debugging                    8.920000   
                Essay Drafting                    9.211538   
                Exam Prep                         8.586207   
                Literature Review                 7.958333   
GitHub Copilot  Brainstorming                     8.126984   
                Code Debugging                    7.406780   
                Essay Drafting                    8.677966   
                Exam Prep                         8.813559   
                Literature Review                 9.070423   
Perplexity      Brainstorming                     9.648148   
                Code Debugging                    8.706897   
                Essay Drafting                    8.791667   
                Exam Prep                         8.500000   
                Literature Review                 8.666667   

                                   Task_Frequency_Daily  
Primary_AI_Tool Main_Usage_Case                          
ChatGPT         Brainstorming                        52  
                Code Debugging                       49  
                Essay Drafting                       63  
                Exam Prep                            69  
                Literature Review                    55  
Claude          Brainstorming                        52  
                Code Debugging                       48  
                Essay Drafting                       55  
                Exam Prep                            68  
                Literature Review                    74  
Gemini Pro      Brainstorming                        65  
                Code Debugging                       50  
                Essay Drafting                       52  
                Exam Prep                            87  
                Literature Review                    48  
GitHub Copilot  Brainstorming                        63  
                Code Debugging                       59  
                Essay Drafting                       59  
                Exam Prep                            59  
                Literature Review                    71  
Perplexity      Brainstorming                        54  
                Code Debugging                       58  
                Essay Drafting                       72  
                Exam Prep                            58  
                Literature Review                    60